absl/types/any.h

```diff
@@ -382,10 +379,20 @@ class any {
    public:
     virtual ~ObjInterface() = default;
     virtual std::unique_ptr<ObjInterface> Clone() const = 0;
-    virtual size_t type_id() const noexcept = 0;
+    virtual const void* ObjTypeId() const noexcept = 0;
 #if ABSL_ANY_DETAIL_HAS_RTTI
     virtual const std::type_info& Type() const noexcept = 0;
 #endif  // ABSL_ANY_DETAIL_HAS_RTTI
+
+    // Note that on 64-bit (unix) systems size_t is 64-bit while int is 32-bit and
+    // the compiler will happily and quietly assign such a 64-bit value to a
+    // 32-bit integer. While a client should never do that it SHOULD still be safe,
+    // assuming the BSS segment doesn't span more than 4GiB.
+    size_t type_id() const noexcept {
```

Comment:
We're casting a pointer to a integral type. I'm suspect we should use `uintptr_t`.

Reply:
This entire function can simply be removed as it is no longer used.